### ISSUE TITLE
401 error when opening the powerbi report

### ISSUE THREAD

**Issue by icemokka** [taarskog/crm-powerbi-viewer#49]
Hi,

I used the beta version 1.2 solution and follow the setup guide.
I see my reports in the solution but when I preview them I get no data, only empty frames.

![image](https://user-images.githubusercontent.com/28389613/48558405-80c1ab00-e8e9-11e8-9326-f04caef91054.png)

Looking at the console (verbose mode), it validates my login via DAL but throws an error on a wabi-uk resource.

![image](https://user-images.githubusercontent.com/28389613/48558519-bebecf00-e8e9-11e8-986b-a2819ae3c76a.png)

Any idea's ?

Regards,
Sven Peeters

**Comment by taarskog** [comment]
First try to log in to Power BI directly (https://app.powerbi.com) and make sure you can see the report there. Also make sure you are properly licensed (pro) #23 

Then verify that all required permissions are set on the App in Azure AD (see guide for details).

If still no luck edit config and increase log levels on both the viewer and adal. Might be you get more info to help investigate this.

**Comment by icemokka** [comment]
Hi Trond,

I can see the report correctly in PowerBI, I'm a licenced Power BI Pro user.
I followed the guide with care, the logs I included are already 'verbose'
setting.

Any other suggestions?

**Comment by taarskog** [comment]
The error is not from crm-powerbi-viewer but directly from power bi. Thus it is most likely related to either privileges set in the Azure AD App or in power bi.

1. Do you get the same message on all types of visuals (reports, dashboards, tiles)?
2. Is it the same experience on both personal workspace and groups?
3. Are you using the preview workspaces?
4. What data source are you trying to show data from? (SQL Server, Dynamics 365 OP/CE/NAV, Google Analytics, ...)

Please also share your config file.

**Comment by icemokka** [comment]
Hi Trond,

I have it on other people's power bi reports ( shared with me ) as wel as
my own power bi dashboards.
The content can be odata source or connection to an azure db (with
replicated data on it)
All reports are showing just fine in powerbi.

here is my config

var customConfig = {
	///////////////////////////////
	// ITEMS THAT MUST BE CHANGED
	///////////////////////////////

	// See documentation on how to register the application in Azure AD
and get a client id (remember to enclose with quotes - see sample).
	// Sample: auth_client_id: "78d86cb0-fcf1-44aa-8e8f-30e70770e9f4",
	auth_client_id: "ac531cf5-a447-468a-90a6-a0498b395250",


	///////////////////////////////////////////
	// CAN USUALLY BE LEFT UNCHANGED
	// (unless specified in the documentation)
	///////////////////////////////////////////

    // How to perform authentication - valid values are "inline" or
"popup". Default is "inline" (rember to enclose with quotes).
    // You should typically use "inline" on Dynamics 365 Online with
users located in your Azure AD and "popup" when on-premise (might also
be required if you have invited external parties to your Azure AD).
	auth_mode: null,

    // Auto-refresh access - may cause page reload if token cannot be
updated silently [valid values are true or false]. Default is true.
	auto_refresh_token: null,

	// Where to cache tokens. Valid values are 'sessionStorage' and
'localStorage'. Default is 'sessionStorage'.
	auth_cache_location: null,

	// Logging of Adal authentication process [Valid values are 0-3]
(0=ERROR, 1=WARNING, 2=INFO, 3=VERBOSE). Default is 0.
	auth_log_level: 3,

	// Log level for Power BI Viewer [Valid values are 0-3] (0=ERROR,
1=WARNING, 2=INFO, 3=DEBUG). Default is 0.
	// (is not affected by and does not affect auth_log_level)
	log_level: 3,

	// Array of custom scripts to load. Functions in these scripts can be
referenced for report filtering etc.
	// Sample: custom_scripts: ["../../new_myfilter?ver=1",
"../../demo_/powerbi/filters/account.js?ver=1"]
    custom_scripts: null
};

**Comment by taarskog** [comment]
1. Just to cover the basics - are Power BI and Dynamics 365 CE both on the same tenant?
2. Can you in Azure AD on app registration give **Power BI Service** permissions to all permissions that do not require admin? (remember to **grant permissions afterwards**).
3. Didn't catch if you answered Q1 and Q3 above:
   1. On your personal workspace do you get the same message on all types of visuals, i.e. reports, dashboards, and tiles?
   1. Are the visuals hosted by preview workspaces (or the "old" type of workspaces)?

**Comment by icemokka** [comment]
Hi Trond,

Yes, got the same issue on all reports.
Don't even know what preview dashboard is, so I assume I don't use it.

There are the permission on the app :

[image: image.png]

**Comment by taarskog** [comment]
Hi Sven,

Your image didn't get through to GitHub ehwn responding by email - seems you need to add from the GitHub website.

The image below shows you how to identify reports, dashboards and tiles. If you have more than reports please try to preview these,
 
![](http://crm-powerbi-viewer.heiigjen.com/assets/images/v1.0/crm-config/config-overview.png)

**Comment by icemokka** [comment]
Hi Trond,

Now that you mention this, I can only see dashboard, there are no reports
visible to me in your solution window.
I found a separate permission for reports on the portal, which I added now.
Everything is working fine now

I can assure you that I followed your procedure correctly (simplified flow).
Maybe check it yourself and extra step to check permissions after the app
has been created.

And your help was greatly appreciated, this kind of support (on a free and
great product) is very rare !!!!!

Regards,
Sven
